### PR TITLE
reverse token and endpoint priority - fixes #16

### LINKED ISSUE #16: Reversing priority of token and endpoint environment variables
Started as a [line comment](https://github.com/sindresorhus/gh-got/commit/1fc6d0265d03a4cf0982464d458db7d7872d9d21#commitcomment-17035522) but let's make it an issue and fix/close it.

The readme says clearly:
> Can be overridden globally with the GITHUB_TOKEN environment variable.

Same for the endpoint.

Not sure what the logic was, but I think it would be preferable to use the environment variables only when no applicable argument is passed. An application could have a default token (the env.) but for whatever reason use another token on specific calls.

The module, readme and test would have to be updated. It could be considered a major change since it's reversing its meaning.

### PR DESCRIPTION
**Pull request by SamVerschueren**
`options.token` is higher priority as discussed in #16.

--------

`options.endpoint` also has a higher priority then `process.env.GITHUB_ENDPOINT`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function ghGot(path, opts) {
 	}, opts.headers);
 
 	const env = process.env;
-	const token = env.GITHUB_TOKEN || opts.token;
+	const token = opts.token || env.GITHUB_TOKEN;
```
Comment (SamVerschueren) [inline index.js]: Might be cleaner if we did the following

```js
opts = Object.assign({
 	json: true, 
 	endpoint: 'https://api.github.com/',
    token: process.env.GITHUB_TOKEN
}, opts);
```

File: index.js [hunk 2]
```diff
@@ -6,15 +6,19 @@ function ghGot(path, opts) {
...
+	}, opts);
 
 	opts.headers = Object.assign({
 		'accept': 'application/vnd.github.v3+json',
 		'user-agent': 'https://github.com/sindresorhus/gh-got'
 	}, opts.headers);
 
-	const env = process.env;
-	const token = env.GITHUB_TOKEN || opts.token;
+	const token = opts.token || env.GITHUB_TOKEN;
```
Comment (SamVerschueren) [inline index.js]: Might be cleaner of we did this at line 11.

```js
opts = Object.assign({
    json: true, 
    endpoint: env.GITHUB_ENDPOINT ? env.GITHUB_ENDPOINT.replace(/[^/]$/, '$&/') : 'https://api.github.com/',
    token: process.env.GITHUB_TOKEN
}, opts);
```

Comment (SamVerschueren) [inline index.js]: Just did it :)

Comment (kevva) [inline index.js]: :+1:

I guess it's no big deal but maybe we should remove `opts.endpoint` and `opts.token` before sending them to `got`.

**Comment by sindresorhus** [comment]
Thank you Sam :)

**Comment by SamVerschueren** [comment]
You're welcome :)

**Comment by SamVerschueren** [comment]
And thanks for merging ;)

**Pull request merged**